File: cpp/hamming/hamming.cpp
#include "hamming.h"
#include <iostream>
#include <stdexcept>

using namespace std;

namespace hamming {

int compute(const string& dna1, const string& dna2){
   if(dna1.length() != dna2.length()){
     throw std::domain_error("Strings are different lengths");
    }
    int hammingNum = 0;
    for(unsigned int i = 0; i < dna1.length(); i++){
      if(dna1[i] != dna2[i]){
	hammingNum++;
      }
    }
    return hammingNum;
  }  
}  // namespace hamming

File: go/hamming/hamming.go
//Package hamming implements the Distance method
package hamming

import (
	"errors"
)

//Distance takes two DNA strings and returns
//the hamming number.
func Distance(a, b string) (int, error) {
	hammingNumber := 0

	if len(a) != len(b) {
		return 0, errors.New("Strings are different lengths")
	}

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			hammingNumber++
		}
	}
	return hammingNumber, nil

}
